File: apps/frontend/src/types/order.ts
/**
 * Dot-Store V2.2 订单相关类型定义
 */

/**
 * 订单类型枚举
 */
export type OrderType = 'dine_in' | 'take_out' | 'pickup';

/**
 * 支付方式枚举
 */
export type PaymentMethod = 'cash' | 'customer_account' | 'wechat' | 'alipay' | 'mixed';

/**
 * 订单接口 - V2.2
 */
export interface Order {
  id: number;
  user_id: number;
  order_no: string;
  order_type: string;
  amount: string;
  payment_method: string | null;
  customer_account_id: number | null;
  category_id: number | null;
  tags: string[] | null;
  order_metadata: Record<string, unknown> | null;
  note: string | null;
  status: string;
  is_deleted: boolean;
  deleted_at: string | null;
  deleted_by: number | null;
  created_at: string;
  updated_at: string;
  created_by: number;
}

/**
 * 订单项接口 - V2.2新增
 */
export interface OrderItem {
  id: number;
  order_id: number;
  product_name: string;
  quantity: string;
  unit_price: string;
  cost_price: string | null;
  amount: string;
  note: string | null;
  created_at: string;
}

/**
 * 订单详情响应 - V2.2新增
 */
export interface OrderDetail extends Order {
  items: OrderItem[];
}

/**
 * 订单分类接口
 */
export interface OrderCategory {
  id: number;
  user_id: number;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
}

/**
 * 创建订单项参数 - V2.2新增
 */
export interface OrderItemCreateParams {
  product_name: string;
  quantity: number | string;
  unit_price: number | string;
  cost_price?: number | string;
  note?: string;
}

/**
 * 创建订单参数 - V2.2
 */
export interface OrderCreateParams {
  order_type: string;
  amount: number | string;
  payment_method?: string;
  customer_account_id?: number;
  category_id?: number;
  tags?: string[];
  note?: string;
  items?: OrderItemCreateParams[];
}

/**
 * 更新订单参数
 */
export interface OrderUpdateParams {
  amount?: number | string;
  order_type?: string;
  payment_method?: string;
  category_id?: number;
  tags?: string[];
  note?: string;
  status?: string;
}

/**
 * 订单作废请求参数 - V2.2新增
 */
export interface OrderVoidParams {
  reason: string;
}

/**
 * 创建订单分类参数
 */
export interface OrderCategoryCreateParams {
  name: string;
  description?: string;
}

/**
 * 更新订单分类参数
 */
export interface OrderCategoryUpdateParams {
  name?: string;
  description?: string;
}

/**
 * 订单筛选条件
 */
export interface OrderFilters {
  start_date?: string;
  end_date?: string;
  order_type?: string;
  category_id?: number;
  status?: string;
  tags?: string;
  page?: number;
  page_size?: number;
}

/**
 * 订单列表响应
 */
export interface OrderListResponse {
  items: Order[];
  total: number;
  page: number;
  page_size: number;
}

/**
 * 订单汇总 - V2.2新增
 */
export interface OrderSummary {
  total_orders: number;
  total_amount: number;
  by_type: Record<string, { count: number; amount: number }>;
  by_payment: Record<string, { count: number; amount: number }>;
}

/**
 * 订单类型选项
 */
export const ORDER_TYPE_OPTIONS = [
  { value: 'dine_in', label: '堂食' },
  { value: 'take_out', label: '自提' },
  { value: 'pickup', label: '打包' },
];

/**
 * 订单状态选项
 */
export const ORDER_STATUS_OPTIONS = [
  { value: 'completed', label: '已完成' },
  { value: 'voided', label: '已作废' },
  { value: 'cancelled', label: '已取消' },
];

/**
 * 支付方式选项 - V2.2新增
 */
export const PAYMENT_METHOD_OPTIONS = [
  { value: 'cash', label: '现金' },
  { value: 'wechat', label: '微信' },
  { value: 'alipay', label: '支付宝' },
  { value: 'customer_account', label: '会员账户' },
  { value: 'mixed', label: '混合支付' },
];

/**
 * 获取订单类型显示名称
 */
export function getOrderTypeLabel(type: string): string {
  const option = ORDER_TYPE_OPTIONS.find(opt => opt.value === type);
  return option ? option.label : type;
}

/**
 * 获取订单状态显示名称
 */
export function getOrderStatusLabel(status: string): string {
  const option = ORDER_STATUS_OPTIONS.find(opt => opt.value === status);
  return option ? option.label : status;
}

/**
 * 获取支付方式显示名称 - V2.2新增
 */
export function getPaymentMethodLabel(method: string): string {
  const option = PAYMENT_METHOD_OPTIONS.find(opt => opt.value === method);
  return option ? option.label : method || '-';
}

File: apps/api-server/app/schemas/order.py
"""
Dot-Store V2.2 订单数据模式
"""
from datetime import datetime
from decimal import Decimal
from typing import Optional, List, Any
from pydantic import BaseModel, Field, ConfigDict


class OrderType:
    """订单类型枚举"""
    DINE_IN = 'dine_in'
    TAKE_OUT = 'take_out'
    PICKUP = 'pickup'


class PaymentMethod:
    """支付方式枚举"""
    CASH = 'cash'
    CUSTOMER_ACCOUNT = 'customer_account'
    WECHAT = 'wechat'
    ALIPAY = 'alipay'
    MIXED = 'mixed'


class OrderItemCreate(BaseModel):
    """订单项创建模式"""
    product_name: str = Field(..., min_length=1, max_length=128, description="产品名称")
    quantity: Decimal = Field(..., gt=0, description="数量")
    unit_price: Decimal = Field(..., ge=0, description="单价")
    cost_price: Optional[Decimal] = Field(None, ge=0, description="成本价")
    note: Optional[str] = Field(None, description="备注")


class OrderItemResponse(BaseModel):
    """订单项响应模式"""
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(..., description="订单项ID")
    order_id: int = Field(..., description="订单ID")
    product_name: str = Field(..., description="产品名称")
    quantity: Decimal = Field(..., description="数量")
    unit_price: Decimal = Field(..., description="单价")
    cost_price: Optional[Decimal] = Field(None, description="成本价")
    amount: Decimal = Field(..., description="金额")
    note: Optional[str] = Field(None, description="备注")
    created_at: datetime = Field(..., description="创建时间")


class OrderCreate(BaseModel):
    """订单创建模式"""
    order_type: str = Field(..., min_length=1, max_length=32, description="订单类型")
    amount: Decimal = Field(..., gt=0, description="订单金额，必须大于0")
    payment_method: Optional[str] = Field(None, max_length=32, description="支付方式")
    customer_account_id: Optional[int] = Field(None, description="客户账户ID")
    category_id: Optional[int] = Field(None, description="分类ID")
    tags: Optional[List[str]] = Field(None, description="订单标签")
    note: Optional[str] = Field(None, description="备注")
    items: Optional[List[OrderItemCreate]] = Field(None, description="订单项列表")


class OrderUpdate(BaseModel):
    """订单更新模式"""
    amount: Optional[Decimal] = Field(None, gt=0, description="订单金额")
    order_type: Optional[str] = Field(None, min_length=1, max_length=32, description="订单类型")
    payment_method: Optional[str] = Field(None, max_length=32, description="支付方式")
    category_id: Optional[int] = Field(None, description="分类ID")
    tags: Optional[List[str]] = Field(None, description="订单标签")
    note: Optional[str] = Field(None, description="备注")
    status: Optional[str] = Field(None, min_length=1, max_length=32, description="订单状态")


class OrderVoidRequest(BaseModel):
    """订单作废请求"""
    reason: str = Field(..., min_length=1, description="作废原因")


class OrderResponse(BaseModel):
    """订单响应模式"""
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(..., description="订单ID")
    user_id: int = Field(..., description="用户ID")
    order_no: str = Field(..., description="订单编号")
    order_type: str = Field(..., description="订单类型")
    amount: Decimal = Field(..., description="订单金额")
    payment_method: Optional[str] = Field(None, description="支付方式")
    customer_account_id: Optional[int] = Field(None, description="客户账户ID")
    category_id: Optional[int] = Field(None, description="分类ID")
    tags: Optional[List[str]] = Field(None, description="订单标签")
    note: Optional[str] = Field(None, description="备注")
    status: str = Field(..., description="订单状态")
    is_deleted: bool = Field(False, description="是否删除")
    deleted_at: Optional[datetime] = Field(None, description="删除时间")
    deleted_by: Optional[int] = Field(None, description="删除人ID")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    created_by: int = Field(..., description="创建人ID")


class OrderDetailResponse(OrderResponse):
    """订单详情响应模式"""
    items: Optional[List[OrderItemResponse]] = Field(None, description="订单项列表")


class OrderListResponse(BaseModel):
    """订单列表响应模式"""
    items: List[OrderResponse] = Field(..., description="订单列表")
    total: int = Field(..., description="总数")
    page: int = Field(1, description="当前页码")
    page_size: int = Field(10, description="每页数量")


class OrderSummary(BaseModel):
    """订单汇总"""
    total_orders: int = Field(0, description="总订单数")
    total_amount: Decimal = Field(0, description="总金额")
    by_type: dict = Field(default_factory=dict, description="按类型统计")
    by_payment: dict = Field(default_factory=dict, description="按支付方式统计")


class OrderCategoryCreate(BaseModel):
    """订单分类创建模式"""
    name: str = Field(..., min_length=1, max_length=64, description="分类名称")
    description: Optional[str] = Field(None, max_length=256, description="分类描述")


class OrderCategoryUpdate(BaseModel):
    """订单分类更新模式"""
    name: Optional[str] = Field(None, min_length=1, max_length=64, description="分类名称")
    description: Optional[str] = Field(None, max_length=256, description="分类描述")


class OrderCategoryResponse(BaseModel):
    """订单分类响应模式"""
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(..., description="分类ID")
    user_id: int = Field(..., description="用户ID")
    name: str = Field(..., description="分类名称")
    description: Optional[str] = Field(None, description="分类描述")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")


class OrderFilters(BaseModel):
    """订单筛选条件"""
    start_date: Optional[datetime] = Field(None, description="开始日期")
    end_date: Optional[datetime] = Field(None, description="结束日期")
    order_type: Optional[str] = Field(None, description="订单类型")
    category_id: Optional[int] = Field(None, description="分类ID")
    tags: Optional[List[str]] = Field(None, description="标签筛选")
    status: Optional[str] = Field(None, description="订单状态")
    page: int = Field(1, ge=1, description="页码")
    page_size: int = Field(10, ge=1, le=100, description="每页数量")
